test/ctia/http/generative/fulltext_search_test.clj

```diff
@@ -178,55 +190,86 @@
-                           (let [res (-> unparsed :parsed-body)]
-                             (is (= 1 (count res)))
-                             (is "first incident" (-> res first :short_description))
-                             (is "Lorem Ipsum Test Incident" (-> res first :title))))}
+       :query-params {:query "\"first\" \"Ipsum\""}
+       :data-gen bulk
+       :check (fn [{:keys [out]}]
+                (let [res (-> out :parsed-body)]
+                  (is (= 1 (count res)))
+                  (is "first incident" (-> res first :short_description))
+                  (is "Lorem Ipsum Test Incident" (-> res first :title))))}
       {:test-description (str "searching in non-default search fields that "
                               "contain the pattern, shall not return any results")
-       :query-params     {:query "white"}
-       :bulk-gen         bulk
-       :check            (fn [_ _ _ res] (is (-> res :parsed-body empty?)))}
+       :query-params {:query "white"}
+       :data-gen bulk
+       :check (fn [{:keys [out]}] (is (-> out :parsed-body empty?)))}
 
       {:test-description "searching in a nested default field, yields results"
-       :query-params     {:query "\"lacinia purus\""}
-       :bulk-gen         (gen/fmap (fn [examples]
-                                     (update examples :sightings
-                                             (fn [sightings]
-                                               (apply utils/update-items sightings
-                                                      [;; update only the first record, leave the rest unchanged
-                                                       #(assoc-in % [:observables 0 :value] "lacinia purus")]))))
-                                   (bulk-gen-for :sightings))
-       :check            (fn [_ _ _ unparsed]
-                           (let [res (-> unparsed :parsed-body)]
-                            (is (= 1 (count res)))
-                            (is (-> res first :observables first :value (= "lacinia purus")))))})]))
+       :query-params {:query "\"lacinia purus\""}
+       :data-gen (gen/fmap (fn [examples]
+                             (update examples :sightings
+                                     (fn [sightings]
+                                       (apply utils/update-items sightings
+                                              [;; update only the first record, leave the rest unchanged
+                                               #(assoc-in % [:observables 0 :value] "lacinia purus")]))))
+                           (bulk-gen-for :sightings))
+       :check (fn [{:keys [out]}]
+                (let [res (-> out :parsed-body)]
+                  (is (= 1 (count res)))
+                  (is (-> res first :observables first :value (= "lacinia purus")))))})]
+
+   [{:test-description (str "split_on_whitespace deprecated in ES6, "
+                            "query expected to work differently in 5 and 7")
```

Comment:
it's not expected to work differently in ES5 and ES7, this is the issue :-)
We do want the same behavior :-)
However the compromise could be to accept not filtering with ES5 and properly implement it for ES7.